src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -1050,17 +1050,13 @@ TEST_CASE("custom_json_data: handling invalid is_shape_tensor")
   }
   SUBCASE("batching - no shm")
   {
-    // FIXME: TMA-765
```

Comment:
Can you explain again what's happening here? Was this an existing bug before our big refactor? And you fixed it with this PR?

Or was this a bug introduced by our big refactor and you're fixing it here?

Reply:
This was never a bug, and this change does not resolve TMA-765 (TMA-765 tags are still in multiple places in the repo)

expect_thread_failure was for cases where we expected a failure during the thread execution (issuing requests)
expect_init_failure was for cases where we expected a failure during init before the thread ran

There are input data errors that we have to catch. For SHM case, we caught them correctly in init. For non-shm, we used to catch them in the thread when we would populate the data for a given request. Now, for non-shm, we catch them in init the same way that shm does.